src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1478,6 +1478,11 @@ GetRTEIdentity(RangeTblEntry *rte)
 	Assert(IsA(rte->values_lists, IntList));
 	Assert(list_length(rte->values_lists) == 1);
 
+	if (rte->values_lists == NULL)
```

Comment:
This check should happen before Asserts. In fact GetRTEIdentity functions is never called. Consider removing the function too.